contractcourt/htlc_timeout_resolver_test.go

```diff
@@ -1324,3 +1324,95 @@ func testHtlcTimeout(t *testing.T, resolution lnwallet.OutgoingHtlcResolution,
 		_ = runFromCheckpoint(t, ctx, checkpoints[i+1:])
 	}
 }
+
+// TestIsPreimageSpend tests `isPreimageSpend` can successfully detect a
+// preimage spend based on whether the commitment is local or remote.
+func TestIsPreimageSpend(t *testing.T) {
+	t.Parallel()
+
+	dummyBytes := []byte{0}
+	preimageBytes := bytes.Repeat([]byte{1}, lntypes.HashSize)
+
+	// witnessSpendLocal is the remote's witness used to spend the HTLC via
+	// the second stage success tx path on the remote commitment.
+	witnessSpendLocal := wire.TxWitness{
```

Comment:
Shouldn't we call this `witnessSpendRemote`?

Reply:
named it to `witnessSpendSuccess` instead. So what I want to differentiate here is not local vs remote but how the preimage is revealed - either via the second stage success tx, or a direct spend. And in the test case setup we then start paying attention to whether it's a local view or a remote view.